Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -2058,326 +2058,364 @@ namespace OpenInfraPlatform
 			}
 
 			void convertIfcHalfSpaceSolid(
-				const std::shared_ptr<typename IfcEntityTypesT::IfcHalfSpaceSolid>& half_space_solid,
+				const EXPRESSReference<typename IfcEntityTypesT::IfcHalfSpaceSolid>& halfSpaceSolid,
 				const carve::math::Matrix& pos,
 				std::shared_ptr<ItemData> itemData,
 				const std::shared_ptr<ItemData>& otherOperand
 			) const noexcept(false)
-			{					
+			{
+				// TODO: complete this function + refactoring 
 				/*
 				//ENTITY IfcHalfSpaceSolid SUPERTYPE OF(ONEOF(IfcBoxedHalfSpace, IfcPolygonalBoundedHalfSpace))
-				std::shared_ptr<typename IfcEntityTypesT::IfcSurface> base_surface = half_space_solid->BaseSurface.lock();
-
-				double length_factor = UnitConvert()->getLengthInMeterFactor();
+				EXPRESSReference<typename IfcEntityTypesT::IfcSurface> base_surface = halfSpaceSolid->BaseSurface;
 
 				// base surface
-				std::shared_ptr<typename IfcEntityTypesT::IfcElementarySurface> elem_base_surface =
-					std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcElementarySurface>(base_surface);
-				if (!elem_base_surface)
-				{
-					BLUE_LOG(warning) << "The base surface shall be an unbounded surface (subtype of IfcElementarySurface)";
-					return;
-				}

```

Comment:
This should not be handled as currently proposed - it is only a check of the attribute, that it is of correct type.

Should be:

```
if( ! base_surface.template isOfType<typename IfcEntityTypesT::IfcElementarySurface>() )
    throw ...
```

With throwing inconsistent modelling exception...

Reply:
Resolved: c3eef4a